### PR TITLE
executor: Support the statement of 'set password = ... ' 

### LINKED ISSUE #2153: Set password without  user information
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
mysql> set password = "1";

2. What did you expect to see?
Query OK, 0 rows affected (0.00 sec)

3. What did you see instead?
ERROR 2013 (HY000): Lost connection to MySQL server during query

### PR DESCRIPTION
**Pull request by zimulala**
Fix #2153 and it helps to solve issue #2073

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @hanfei1991 @coocood @tiancaiamao

**Comment by coocood** [comment]
LGTM

File: executor/executor_simple.go [hunk 1]
```diff
@@ -237,11 +237,25 @@ func userExists(ctx context.Context, name string, host string) (bool, error) {
 }
 
 func (e *SimpleExec) executeSetPwd(s *ast.SetPwdStmt) error {
-	// TODO: If len(s.User) == 0, use CURRENT_USER()
+	if len(s.User) == 0 {
+		vars := e.ctx.GetSessionVars()
+		s.User = vars.User
+		if len(s.User) == 0 {
+			return errors.New("Session error is empty")
```
Comment (hanfei1991) [inline executor/executor_simple.go]: When the user would be empty?

File: executor/executor.go [hunk 2]
```diff
@@ -67,26 +67,28 @@ var (
...
-	ErrRowKeyCount     = terror.ClassExecutor.New(CodeRowKeyCount, "Wrong row key entry count")
-	ErrPrepareDDL      = terror.ClassExecutor.New(CodePrepareDDL, "Can not prepare DDL statements")
+	ErrUnknownPlan     = terror.ClassExecutor.New(codeUnknownPlan, "Unknown plan")
+	ErrPrepareMulti    = terror.ClassExecutor.New(codePrepareMulti, "Can not prepare multiple statements")
+	ErrStmtNotFound    = terror.ClassExecutor.New(codeStmtNotFound, "Prepared statement not found")
+	ErrSchemaChanged   = terror.ClassExecutor.New(codeSchemaChanged, "Schema has changed")
+	ErrWrongParamCount = terror.ClassExecutor.New(codeWrongParamCount, "Wrong parameter count")
+	ErrRowKeyCount     = terror.ClassExecutor.New(codeRowKeyCount, "Wrong row key entry count")
+	ErrPrepareDDL      = terror.ClassExecutor.New(codePrepareDDL, "Can not prepare DDL statements")
+	ErrPasswordNoMatch = terror.ClassExecutor.New(CodePasswordNoMatch, "Can't find any matching row in the user table")
```
Comment (hanfei1991) [inline executor/executor.go]: This error seems to represent there is no user that matches the username. Why you call it "Password no match"?

**Comment by hanfei1991** [comment]
LGTM

File: executor/executor.go (hunk 2 above)
Comment (zimulala) [inline executor/executor.go]: This is according to MySQL

**Pull request merged**